modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -134,14 +137,14 @@ public class RocksDbMvPartitionStorage implements MvPartitionStorage {
     /** Thread-local direct buffer instance to read keys from RocksDB. */
     private static final ThreadLocal<ByteBuffer> MV_KEY_BUFFER = withInitial(() -> allocateDirect(MAX_KEY_SIZE).order(KEY_BYTE_ORDER));
 
-    /** Thread-local write batch for {@link #runConsistently(WriteClosure)}. */
-    private static final ThreadLocal<WriteBatchWithIndex> WRITE_BATCH = new ThreadLocal<>();
-
     /** Thread-local on-heap byte buffer instance to use for key manipulations. */
     private static final ThreadLocal<ByteBuffer> HEAP_KEY_BUFFER = withInitial(
             () -> ByteBuffer.allocate(MAX_KEY_SIZE).order(KEY_BYTE_ORDER)
     );
 
+    /** Thread-local write batch for {@link #runConsistently(WriteClosure)}. */
+    private final ThreadLocal<WriteBatchWithIndex> threadLocalWriteBatch = new ThreadLocal<>();
```

Comment:
Why not `static` ?

Reply:
static would not allow us to operate with multiple partitions from the same thread. No one would do that, but technically non-static field is a correct solution here